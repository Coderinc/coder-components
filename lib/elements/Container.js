import styled from 'styled-components';

export default styled.div`
  box-sizing: border-box;
  margin: 0 auto;
  width: 100%;
  max-height: 100%;
  max-width: ${props => {
    if (props.small) {return '40rem';}
    else if (props.large) {return '75rem';}
    else if (props.extraLarge) {return '100rem';}
    return '50rem';
  }};
`;

