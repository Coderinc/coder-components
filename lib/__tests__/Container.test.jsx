import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Container from '../elements/Container';

describe('Container', () => {
  it('renders correct', () => {
    const component = renderer.create(<Container />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('sets proper size', () => {
    const component = renderer.create(<Container large />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
