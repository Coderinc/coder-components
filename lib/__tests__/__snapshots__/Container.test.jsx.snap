// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correct 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0 auto;
  padding: 1rem;
  width: 100%;
  max-height: 100%;
  max-width: 50rem;
}

<div
  className="c0"
/>
`;

exports[`Container sets proper size 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0 auto;
  padding: 1rem;
  width: 100%;
  max-height: 100%;
  max-width: 75rem;
}

<div
  className="c0"
/>
`;
